packages/abi-typegen/test/fixtures/templates/predicate-with-configurable/main.hbs

```diff
@@ -90,27 +88,22 @@ const _abi = {
       "offset": 872
     }
   ]
-}
-
-const _bin = '0x000'
+};
 
-export const MyPredicateAbi__factory = {
+const bytecode = '0x000';
 
-  abi: _abi,
-  bin: _bin,
+export class MyPredicate extends Predicate<MyPredicateInputs> {
 
-  createInstance(provider: Provider, predicateData?: MyPredicateAbiInputs, configurables?: MyPredicateAbiConfigurables) {
+  static readonly abi = abi;
+  static readonly bytecode = bytecode;
 
-    const predicate = new Predicate<MyPredicateAbiInputs>({
-      bytecode: _bin,
-      abi: _abi,
+  constructor(provider: Provider, predicateData?: MyPredicateInputs, configurables?: MyPredicateConfigurables) {
```

Comment:
Ideally, this constructor should follow the constructor of the `Predicate` class.
```ts
export type PredicateParams<T = InputValue[]> = {
  bytecode: BytesLike;
  provider: Provider;
  abi?: JsonAbi;
  inputData?: T;
  configurableConstants?: { [name: string]: unknown };
};
```

Reply:
Ideally, this constructor should follow the constructor of `Predicate`, just with certain values optional because they're already set internally. Note that in the example below I have type-specified the configurables as well so that `PredicateParams` can be reused cleaner. The `C` type parameter doesn't currently exist in `PredicateParams`.

```ts
// predicate constructor
export type PredicateParams<T = InputValue[], C = { [name: string]: unknown }> = {
  bytecode: BytesLike;
  provider: Provider;
  abi?: JsonAbi;
  inputData?: T;
  configurableConstants?: C;
};

export type TypegenPredicateParams = Omit<
  PredicateParams<MyPredicateInput, MyPredicateConfigurables>,
  'bytecode'
>;
```

Reply:
@petertonysmith94 Great idea. Any idea of how much work that would be?

If you're keen to jump in and create another PR for this one, that'd be excellent. 😄

Reply:
@nedsalk Yep, and if we use named params, we could still pick/omit/partial as we wish, right?

Using named parameters in both would also benefit the base Predicate class I suppose.

Thoughts?

Reply:
Reconsider this after what @danielbate said [here](https://github.com/FuelLabs/fuels-ts/pull/2874#pullrequestreview-2212513455) — let's try to keep "extras" out of the scope.

Since we're breaking the generated API, this can be easy on that end. The `Predicate` constructor doesn't need to change now. People are always using typegend Predicates anyway.

Reply:
@danielbate What do you think of this case in particular?

Reply:
@nedsalk I would have favoured that approach, just popping the question :) 

What would peoples thoughts on changing:
- `inputData` -> `data`
- `configurableConstants` -> `configurables`

```ts
// predicate constructor
export type PredicateParams<T = InputValue[], C = { [name: string]: unknown }> = {
  bytecode: BytesLike;
  provider: Provider;
  abi?: JsonAbi;
  data?: T;
  configurables?: C;
};
```

Reply:
@arboleya I'm happy to take on the changes :)

Reply:
I am on the side of this idea for the typegend Predicate, while keeping the base one intact.

```ts
constructor(provider: Provider, opts?: {
    data?: MyPredicateInputs,
    configurables?: MyPredicateConfigurables
  }) {
```

Reply:
This is a valuable change, worthwhile for [the same reasons](https://github.com/FuelLabs/fuels-ts/pull/1826) that we changed the base predicate class constructor. Also DX between the base class and the typegen class is a good point. Especially if we've now got parity in how they are instantiated.

Reply:
> Yep, and if we use named params, we could still pick/omit/partial as we wish, right?

Do you mean named params as in `[key1: value, key2: value]` or the object params? If you mean the first case, you can't work with the key names in typescript, they serve only a visual purpose for LSPs. You can manipulate the elements of the array, though, as they're just regular types.
If you mean the second case (object params), then that's true and what the solution I provided shows.  

> I am on the side of this idea for the typegend Predicate, while keeping the base one intact.

IMO the `Predicate` class's constructor is good - it's _one_ params object taking in everything. The rationale behind the params object is explained in #1826. I'm apprehensive of having different APIs between the generated and "base" code because changes to the base code's types won't be reflected in the typegend code. Ideally, typegen should be augmenting the base class constructors, not modifying them.

Reply:
Good, I think we are talking about the same thing. I'm in favor of having one param object taking in everything as well, but both in the base Predicate class as in the generated one. The difference in the generated one is that `abi` and `bytecode` should be optional and default to the ones generated with the file.

```ts
super({
  abi: opts.abi ?? abi,
  bytecode: opts.bytecode ?? bytecode,
  provider,
  data,
  configurable,
});
```

Makes sense?

Reply:
I've stuck with `configurableConstants` as it's used in `Script` and `Contract`, and I believe they should all be in sync.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2876

Reply:
Aside from this, all tests are passing, and the PR is ready for review. 🥳 

Minor trick I added in this commit for auto-updating fixtures, think of it like snapshots:
 - [Adding option to auto-update fixtures (`21b979f`)](https://github.com/FuelLabs/fuels-ts/pull/2824/commits/21b979f14c634d5421fc78b12b4e5b8f7b699fad)